retry/exp_backoff.go

```diff
@@ -0,0 +1,114 @@
+package retry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/simplesurance/cfdns/logs"
+)
+
+// ExpBackoff executes the provided function, retrying it with exponential
+// back-off if it fails. Context cancellation is respected. The callback
+// function can mark an error as permanent by wrapping it will
+// retry.PermanentError, in which case no more attempts will be made.
+//
+// firstDelay specifies how long it should wait before trying again before
+// retrying the first time. On each retry the delay will be multiplied by
+// the provided factor, but will not be longer than maxDelay.
+//
+// attempts indicates how many times the function call be invoked. The value
```

Comment:
optional:

```suggestion
// attempts indicates how many times the function is invoked. The value
```

Alternatively, another good parameter name would be: `max_tries`